doc/source/reference/c-api/array.rst

```diff
@@ -3213,19 +3215,31 @@ the C-API is needed then some additional steps must be taken.
     header file as long as you make sure that NO_IMPORT_ARRAY is
     #defined before #including that file.
 
-    Internally, these #defines work as follows:
+    If the single extension module not only consists of multiple
+    compilation units, but also spans multiple binaries, instead of
+    invoking ``import_array`` in every binary, you can leverage the
+    ``PY_ARRAY_ATTRIBUTE`` macro. Internally, the C-API is actually
```

Comment:
I feel this should be clear about the hidden visibility.  The details can be in the bullet points below.